orocos_kdl/src/frames.cpp

```diff
@@ -413,15 +413,13 @@ double Rotation::GetRotAngle(Vector& axis,double eps) const {
         return angle; // return 180 deg rotation
     }
 
-    // If the matrix is slightly non-orthogonal, `f` may be out of the (-1, +1) range.
-    // Therefore, clamp it between those values to avoid NaNs.
     double f = (data[0] + data[4] + data[8] - 1) / 2;
-    angle = acos(std::max(-1.0, std::min(1.0, f)));
 
     x = (data[7] - data[5]);
     y = (data[2] - data[6]);
     z = (data[3] - data[1]);
     axis = KDL::Vector(x, y, z);
+    angle = atan2(axis.Norm()/2,f);
```

Comment:
Not directly related, but I think the tests for singularities above can be simplified significantly if they would be based on the pre-computed values for the general case
```cpp
double c = (data[0] + data[4] + data[8] - 1) / 2;    // == f == cos(angle)
```
and
```cpp
double s = axis.Norm()/2;    // == sin(angle)
```
Only the `atan(s,c)` and `axis.Normalize()` calls would fail for the corner cases `angle == 0` or `angle == 180`.

So
```cpp
    if ((std::abs(data[1] - data[3]) < epsilon)
        && (std::abs(data[2] - data[6])< epsilon)
        && (std::abs(data[5] - data[7]) < epsilon))
```
is equivalent to
```cpp
    if (s < epsilon)  // angle is close to 0 or 180 degrees
```
and the inner check
```cpp
        if ((std::abs(data[1] + data[3]) < epsilon2)
            && (std::abs(data[2] + data[6]) < epsilon2)
            && (std::abs(data[5] + data[7]) < epsilon2)
            && (std::abs(data[0] + data[4] + data[8]-3) < epsilon2))
```
is equivalent to
```cpp
    if ((std::abs(c - 1) < epsilon)  // cos(angle) is approximately 1 => angle is 0
```

Furthermore
```cpp
double half_sqrt_2 = 0.5 * sqrt(2.0);
```
is unused.